### PR TITLE
Add ILLink.Descriptors.xml for the Avalonia.Controls

### PR DESCRIPTION
**Pull request by maxkatz6**
## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026389-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Controls/ILLink.Descriptors.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<linker>
+  <assembly fullname="Avalonia.Controls">
+    <type fullname="Avalonia.Controls.NativeMenu*" preserve="all"/>
```
Comment (maxkatz6) [inline src/Avalonia.Controls/ILLink.Descriptors.xml:3]:
```suggestion
    <type fullname="Avalonia.Controls.NativeMenuItem" preserve="all"/>
```

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026392-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0026395-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by hez2010** [comment]
How about using `DynamicDependencyAttribute` instead of embedding an ILLink.Descriptors.xml?

**Comment by maxkatz6** [comment]
@hez2010 not sure how to achieve this. I can't set DynamicDependencyAttribute on the NativeMenuItem and other classes. Also https://github.com/dotnet/runtime/issues/37352#issuecomment-644385807

**Comment by maxkatz6** [comment]
@hez2010 or I can try something like this:
```csharp
        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(NativeMenuItem))]
        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(NativeMenuItemBase))]
        public NativeMenuBar()
        {

        }
```

**Comment by maxkatz6** [comment]
Closing here and will continue with https://github.com/AvaloniaUI/Avalonia/pull/9480

**Pull request closed without merging**